libr/bin/bin.c

```diff
@@ -962,19 +962,27 @@ static void list_xtr_archs(RBin *bin, int mode) {
 			machine = xtr_data->metadata->machine;
 			bits = xtr_data->metadata->bits;
 			switch (mode) {
-			case 'q':
+			case 'q': // "iAq"
 				bin->cb_printf ("%s\n", arch);
 				break;
-			case 'j':
-				bin->cb_printf (
-					"%s{\"arch\":\"%s\",\"bits\":%d,"
-					"\"offset\":%" PFMT64d
-					",\"size\":%" PFMT64d
-					",\"machine\":\"%s\"}",
-					i++ ? "," : "", arch, bits,
-					xtr_data->offset, xtr_data->size,
-					machine);
+			case 'j': { // "iAj"
+				PJ * pj = pj_new ();
+				pj_o (pj);
+				pj_a (pj);
+				pj_ks (pj, "arch", arch);
+				pj_ki (pj, "bits", bits);
+				pj_ki (pj, "offset", xtr_data->offset);
+				pj_kn (pj, "size", xtr_data->size);
+				if (machine) {
+					pj_ks (pj, "machine", machine);
+				}
+				pj_end (pj);
+				pj_end (pj);
+				char *s = pj_drain (pj);
+				r_cons_println (s);
```

Comment:
```
../libr/bin/bin.c:982: undefined reference to `r_cons_println'
```
This should be:
```c
bin->cb_printf ("%s\n", pj_string (pj));
pj_free (pj);
```

Reply:
Thank you @thestr4ng3r, fixed it now.
Marking this conversation as resolved.